---
title: "diannDataProcessing"
author: "Christopher Hughes"
format: html
editor: visual
---

## Details

This document describes processing of proteomics data from DIANN to compile a final protein set.

## Setting up the environment

These are packages you will need for this notebook. For exact versions used, please refer to the session info at the bottom of this notebook.

```{r}
#| message: false
#| warning: false
##########################################################################################
library('tidyverse')
library('arrow')
library('iq')
library('DEqMS')
library('RColorBrewer')
library('ggrepel')
```

I want to set a base directory that we can use as a working space.

```{r}
##########################################################################################
baseRepository = 'E:/requests/1191/240425_1191_orbitrapLumos'
```

## Data processing

Read in data from DIA-NN and process with the IQ package.

```{r}
##########################################################################################
##first we can filter and parse the report data into a better format for IQ
##I had to do some extra stuff here because the sample ID was in a bad format
##if you are using DIA-NN version 1.9 or greater, you need to rename the Fragment.Quant column
#
dia1 = read_parquet(paste(baseRepository, '/diannDia1/report.parquet',sep=''),) %>%
  dplyr::mutate(Run = sub('.*240425_1191_zc_(.*)_dia[123]', '\\1', Run)) %>%
  #dplyr::filter(!grepl(';', Protein.Group)) %>%
  dplyr::mutate(fraction = 1)

#
dia2 = read_parquet(paste(baseRepository, '/diannDia2/report.parquet',sep='')) %>%
  dplyr::mutate(Run = sub('.*240425_1191_zc_(.*)_dia[123]', '\\1', Run)) %>%
 #dplyr::filter(!grepl(';', Protein.Group)) %>%
  dplyr::mutate(fraction = 2)

#
dia3 = read_parquet(paste(baseRepository, '/diannDia3/report.parquet',sep='')) %>%
  dplyr::mutate(Run = sub('.*240425_1191_zc_(.*)_dia[123]', '\\1', Run)) %>%
  #dplyr::filter(!grepl(';', Protein.Group)) %>%
  dplyr::mutate(fraction = 3)

#
raw = rbind(dia1,dia2,dia3)

##save the data
write.table(raw, paste(baseRepository,'/dataProcessing/combinedFilteredReportAnnotated.tsv',sep=''), 
            col.names = TRUE, row.names = FALSE, quote = FALSE, sep = '\t')

##run the IQ analysis
iqResult = process_long_format(paste(baseRepository,'/dataProcessing/combinedFilteredReportAnnotated.tsv',sep=''),
                           output_filename = paste(baseRepository,'/dataProcessing/combinedFilteredReportAnnotatedIq.tsv',sep=''),
                           sample_id = "Run",
                           secondary_id = c("Precursor.Id", "fraction"),
                           primary_id = "Protein.Group",
                           intensity_col = "Precursor.Normalised",
                           intensity_col_sep = NULL,
                           annotation_col = c("Protein.Ids",
                                              "Protein.Names", 
                                              "Genes"),
                           filter_double_less = c("Q.Value" = "0.01",
                                                  "PG.Q.Value" = "0.05", 
                                                  "Lib.Q.Value" = "0.01", 
                                                  "Lib.PG.Q.Value" = "0.01"),
                           peptide_extractor = function(x) gsub("[0-9].*$", "", x))

##do the same for the human index
annotationDb = readRDS(paste(baseRepository, '/databases/2025-04-14-reviewed-UP000005640-spikein.fasta_index_trypticPeps.rds',sep='')) %>%
  dplyr::select(-species) %>%
  dplyr::filter(!grepl('sp\\|V.*',gene))

##parse the IQ result
proData = read_tsv(paste(baseRepository,'/dataProcessing/combinedFilteredReportAnnotatedIq.tsv',sep=''), show_col_types = FALSE) %>%
  dplyr::rename(accession = 'Protein.Group') %>%
  dplyr::mutate(accession = sub('.*\\;','',accession)) %>%
  dplyr::select(-Genes) %>%
  dplyr::left_join(annotationDb)

##annotate the columns
#annoData = read_tsv(paste(baseRepository,'/bmsSampleMatrix.txt',sep=''),show_col_types = FALSE)
#numberOfSamples = sum(grepl('sample',colnames(proData)))
#proDataAnnotated = proData %>%
#  dplyr::select(accession:n_peptides, gene:detectableLength, 
#                sort(factor(colnames(proData)[order(colnames(proData)[6:(numberOfSamples+5)])+5],
#                            levels=c(paste('sample',seq(1,numberOfSamples,1),sep=''))))) %>%
#  dplyr::rename_with(~ annoData$sampleName[1:numberOfSamples], starts_with('sample')) %>%
#  dplyr::arrange(-n_peptides) %>%
#  dplyr::filter(!grepl('CONT',accession))

##save the data
saveRDS(proData, paste(baseRepository, '/dataProcessing/parsedProteinData.rds', sep = ''))
write.table(proData, paste(baseRepository, '/dataProcessing/parsedProteinData.tsv', sep = ''), 
            col.names = TRUE, row.names = FALSE, quote = FALSE, sep = '\t')
```

Perform differential expression analysis on the data.

```{r}
##########################################################################################
##I am going to do one-by-one comparisons
processedData = readRDS(paste(baseRepository, '/dataProcessing/parsedProteinData.rds', sep = '')) %>%
  dplyr::filter(!grepl(';',Protein.Ids))
annotationDb = readRDS(paste(baseRepository, '/databases/2025-04-14-reviewed-UP000005640-spikein.fasta_index_trypticPeps.rds',sep='')) %>%
  dplyr::select(-species) %>%
  dplyr::filter(!grepl('sp\\|V.*',gene)) #removes viral accessions

##rename the columns and setup the comparisons
conditionsToCompare = c('sampleCS')
controlCondition = 'sampleRU'

#
for (i in 1:length(conditionsToCompare)){
  dat = processedData[,c(which(names(processedData) == 'accession'),
                         which(names(processedData) == 'gene'),      
                         which(names(processedData) == paste(controlCondition,'1', sep = '')),
                         which(names(processedData) == paste(controlCondition,'2', sep = '')),
                         which(names(processedData) == paste(controlCondition,'3', sep = '')),
                         which(names(processedData) == paste(conditionsToCompare[i],'1', sep = '')),
                         which(names(processedData) == paste(conditionsToCompare[i],'2', sep = '')),
                         which(names(processedData) == paste(conditionsToCompare[i],'3', sep = '')))]
  dat$samples123 = rowSums(is.na(dat[,3:5]))
  dat$samples456 = rowSums(is.na(dat[,6:8]))
  colnames(dat)[3:5] = c('cont_rep1','cont_rep2','cont_rep3')
  colnames(dat)[6:8] = c('treat_rep1','treat_rep2','treat_rep3')
  #
  datFiltered = as.data.frame(dat) %>%
    dplyr::filter(samples123 < 1 | samples456 < 1) %>%
    dplyr::mutate(cont_rep1 = ifelse(samples123 == 3, min(cont_rep1, na.rm = TRUE), cont_rep1),
                  cont_rep2 = ifelse(samples123 == 3, min(cont_rep2, na.rm = TRUE), cont_rep2),
                  cont_rep3 = ifelse(samples123 == 3, min(cont_rep3, na.rm = TRUE), cont_rep3)) %>%
    dplyr::mutate(treat_rep1 = ifelse(samples456 == 3, min(treat_rep1, na.rm = TRUE), treat_rep1),
                  treat_rep2 = ifelse(samples456 == 3, min(treat_rep2, na.rm = TRUE), treat_rep2),
                  treat_rep3 = ifelse(samples456 == 3, min(treat_rep3, na.rm = TRUE), treat_rep3))
  #
  rownames(datFiltered) = datFiltered$accession
  dat.log = datFiltered %>%
    dplyr::select(-accession, -gene, -samples123, -samples456)
  #dat.log = na.omit(datFiltered)
  #boxplot(dat.log, las = 2, main = 'total protein abundance')
  
  ##make a design table
  cond = as.factor(c(rep('cont',3),
                    rep('treat',3)))
  design = model.matrix(~0+cond)
  colnames(design) = gsub('cond', '', colnames(design))
  
  ##make the contrasts
  x = c('treat-cont')
  contrast = makeContrasts(contrasts = x, levels = design)
  fit1 = lmFit(dat.log, design)
  fit2 = contrasts.fit(fit1, contrasts = contrast)
  fit3 = eBayes(fit2)

  ##add information about PSM counts
  forPsm = processedData
  psm.count.table = data.frame(count = forPsm$n_peptides,
                               row.names = forPsm$accession)
  fit3$count = psm.count.table[rownames(fit3$coefficients),'count']
  fit4 = spectraCounteBayes(fit3)
  #VarianceBoxplot(fit4, n = 30, main = 'total proteome dataset', xlab = 'psm count')
  #VarianceScatterplot(fit4, main = 'total proteome dataset')
  
  ##extract the data
  #head(fit4$coefficients)
  psmAnnotate = dplyr::select(forPsm, accession, gene, n_fragments, n_peptides)
  valuesAnnotate = dplyr::select(datFiltered, accession, gene, samples123, samples456) %>%
    dplyr::rename(controlMissingValues = 'samples123',
                  conditionMissingValues = 'samples456')
  deqmsResults = tibble(outputResult(fit4, coef_col = 1)) %>%
    dplyr::rename(accession = 'gene') %>%
    dplyr::left_join(psmAnnotate) %>%
    dplyr::left_join(annotationDb) %>%
    dplyr::left_join(valuesAnnotate)
  #
  saveRDS(deqmsResults, paste(baseRepository, '/dataProcessing/dataset_deqms_', conditionsToCompare[i], '-', controlCondition, '.rds', sep = ''))
  write.table(deqmsResults, paste(baseRepository, '/dataProcessing/dataset_deqms_', conditionsToCompare[i], '-', controlCondition, '.tsv', sep = ''),
              col.names = TRUE, row.names = FALSE, sep = '\t', quote = FALSE)
  
  ##create a plot of the data
  #create the plot and save it
  goiUp = dplyr::filter(deqmsResults, logFC > 1)$gene[1:20]
  goiDn = dplyr::filter(deqmsResults, logFC < -1)$gene[1:20]
  
  plotData = deqmsResults %>%
    dplyr::mutate(logFC = ifelse(logFC > 6, 6,
                                 ifelse(logFC < -6, -6, logFC))) %>%
    dplyr::mutate(pColors = ifelse(logFC > 0.5, brewer.pal(4,'OrRd')[4],
                                   ifelse(logFC < -0.5, brewer.pal(4,'GnBu')[4], brewer.pal(3,'Greys')[2]))) %>%
    dplyr::mutate(pTextUp = ifelse(gene %in% goiUp, gene, '')) %>%
    dplyr::mutate(pTextDn = ifelse(gene %in% goiDn, gene, ''))
  #
  ggplot(plotData, aes(logFC, -log10(sca.P.Value))) +
    geom_point(size = 0.5, color = plotData$pColors) +
    labs(x = paste('log2(',conditionsToCompare[i],'-',controlCondition,')',sep=''), y = '-log10(p-value)', title = 'Differential abundance') +
    geom_text_repel(label = plotData$pTextUp, min.segment.length = unit(0, 'lines'), nudge_x = -1, nudge_y = 1, max.overlaps = 55000, size = 1, segment.size = 0.15) +
    geom_text_repel(label = plotData$pTextDn, min.segment.length = unit(0, 'lines'), nudge_x = 1, nudge_y = 1, max.overlaps = 55000, size = 1, segment.size = 0.15) +
    scale_x_continuous(limits = c(-6,6), breaks = seq(-20,20,2)) +
    scale_y_continuous(limits = c(0,20), breaks = seq(0,60,5)) +
    geom_vline(xintercept = c(-0.5,0.5), linetype = 'dashed') +
    geom_hline(yintercept = -log10(0.01), linetype = 'dashed') +
    theme_classic()
  ggsave(paste(baseRepository, '/dataProcessing/scatter_deqms_', conditionsToCompare[i], '-', controlCondition, '.pdf', sep = ''),
         height = 2, width = 2, useDingbats = FALSE)
}
```

Read in data from DIA-NN and process with the IQ package.

```{r}
##########################################################################################
##first we can filter and parse the report data into a better format for IQ
##I had to do some extra stuff here because the sample ID was in a bad format
##if you are using DIA-NN version 1.9 or greater, you need to rename the Fragment.Quant column
#
dia1 = read_parquet(paste(baseRepository, '/kidney/diannDia1/report.parquet',sep=''),) %>%
  dplyr::mutate(Run = sub('.*250321_1223_fawcettProteome_(.*)_dia[123]', '\\1', Run)) %>%
  #dplyr::filter(!grepl(';', Protein.Group)) %>%
  dplyr::mutate(fraction = 1)

#
dia2 = read_parquet(paste(baseRepository, '/kidney/diannDia2/report.parquet',sep='')) %>%
  dplyr::mutate(Run = sub('.*250321_1223_fawcettProteome_(.*)_dia[123]', '\\1', Run)) %>%
 #dplyr::filter(!grepl(';', Protein.Group)) %>%
  dplyr::mutate(fraction = 2)

#
dia3 = read_parquet(paste(baseRepository, '/kidney/diannDia3/report.parquet',sep='')) %>%
  dplyr::mutate(Run = sub('.*250321_1223_fawcettProteome_(.*)_dia[123]', '\\1', Run)) %>%
  #dplyr::filter(!grepl(';', Protein.Group)) %>%
  dplyr::mutate(fraction = 3)

#
raw = rbind(dia1,dia2,dia3)

##save the data
write.table(raw, paste(baseRepository,'/kidney/dataProcessing/combinedFilteredReportAnnotated.tsv',sep=''), 
            col.names = TRUE, row.names = FALSE, quote = FALSE, sep = '\t')

##run the IQ analysis
iqResult = process_long_format(paste(baseRepository,'/kidney/dataProcessing/combinedFilteredReportAnnotated.tsv',sep=''),
                           output_filename = paste(baseRepository,'/kidney/dataProcessing/combinedFilteredReportAnnotatedIq.tsv',sep=''),
                           sample_id = "Run",
                           secondary_id = c("Precursor.Id", "fraction"),
                           primary_id = "Protein.Group",
                           intensity_col = "Precursor.Normalised",
                           intensity_col_sep = NULL,
                           annotation_col = c("Protein.Ids",
                                              "Protein.Names", 
                                              "Genes"),
                           filter_double_less = c("Q.Value" = "0.01",
                                                  "PG.Q.Value" = "0.05", 
                                                  "Lib.Q.Value" = "0.01", 
                                                  "Lib.PG.Q.Value" = "0.01"),
                           peptide_extractor = function(x) gsub("[0-9].*$", "", x))

##do the same for the human index
annotationDb = readRDS(paste(baseRepository, '/databases/2025-03-24-reviewed-UP000000589-spikein.fasta_index_trypticPeps.rds',sep='')) %>%
  dplyr::select(-species) %>%
  dplyr::filter(!grepl('sp\\|V.*',gene))

##parse the IQ result
proData = read_tsv(paste(baseRepository,'/kidney/dataProcessing/combinedFilteredReportAnnotatedIq.tsv',sep=''), show_col_types = FALSE) %>%
  dplyr::rename(accession = 'Protein.Group') %>%
  dplyr::mutate(accession = sub('.*\\;','',accession)) %>%
  dplyr::select(-Genes) %>%
  dplyr::left_join(annotationDb)

##annotate the columns
annoData = read_tsv(paste(baseRepository,'/kidney/bmsSampleMatrix.txt',sep=''),show_col_types = FALSE)
numberOfSamples = sum(grepl('sample',colnames(proData)))
proDataAnnotated = proData %>%
  dplyr::select(accession:n_peptides, gene:detectableLength, 
                sort(factor(colnames(proData)[order(colnames(proData)[6:(numberOfSamples+5)])+5],
                            levels=c(paste('sample',seq(as.numeric(sub('sample','',annoData$bmsId[1])),as.numeric(sub('sample','',annoData$bmsId[numberOfSamples])),1),sep=''))))) %>%
  dplyr::rename_with(~ annoData$sampleName[1:numberOfSamples], starts_with('sample')) %>%
  dplyr::arrange(-n_peptides) %>%
  dplyr::filter(!grepl('CONT',accession))

##save the data
saveRDS(proDataAnnotated, paste(baseRepository, '/kidney/dataProcessing/parsedProteinData.rds', sep = ''))
write.table(proDataAnnotated, paste(baseRepository, '/kidney/dataProcessing/parsedProteinData.tsv', sep = ''), 
            col.names = TRUE, row.names = FALSE, quote = FALSE, sep = '\t')
```

Perform differential expression analysis on the data.

```{r}
##########################################################################################
##I am going to do one-by-one comparisons
processedData = readRDS(paste(baseRepository, '/kidney/dataProcessing/parsedProteinData.rds', sep = '')) %>%
  dplyr::filter(!grepl(';',Protein.Ids))
annotationDb = readRDS(paste(baseRepository, '/databases/2025-03-24-reviewed-UP000000589-spikein.fasta_index_trypticPeps.rds',sep='')) %>%
  dplyr::select(-species) %>%
  dplyr::filter(!grepl('sp\\|V.*',gene)) #removes viral accessions

##rename the columns and setup the comparisons
conditionsToCompare = c('KO')
controlCondition = 'WT'

#
for (i in 1:length(conditionsToCompare)){
  dat = processedData[,c(which(names(processedData) == 'accession'),
                         which(names(processedData) == 'gene'),      
                         which(names(processedData) == paste(controlCondition,'1', sep = '')),
                         which(names(processedData) == paste(controlCondition,'2', sep = '')),
                         which(names(processedData) == paste(controlCondition,'3', sep = '')),
                         which(names(processedData) == paste(controlCondition,'4', sep = '')),
                         which(names(processedData) == paste(conditionsToCompare[i],'1', sep = '')),
                         which(names(processedData) == paste(conditionsToCompare[i],'2', sep = '')),
                         which(names(processedData) == paste(conditionsToCompare[i],'3', sep = '')),
                         which(names(processedData) == paste(conditionsToCompare[i],'4', sep = '')))]
  dat$samples123 = rowSums(is.na(dat[,3:6]))
  dat$samples456 = rowSums(is.na(dat[,7:10]))
  colnames(dat)[3:6] = c('cont_rep1','cont_rep2','cont_rep3','cont_rep4')
  colnames(dat)[7:10] = c('treat_rep1','treat_rep2','treat_rep3','treat_rep4')
  #
  datFiltered = as.data.frame(dat) %>%
    dplyr::filter(samples123 < 1 | samples456 < 1) %>%
    dplyr::mutate(cont_rep1 = ifelse(samples123 == 4, min(cont_rep1, na.rm = TRUE), cont_rep1),
                  cont_rep2 = ifelse(samples123 == 4, min(cont_rep2, na.rm = TRUE), cont_rep2),
                  cont_rep3 = ifelse(samples123 == 4, min(cont_rep3, na.rm = TRUE), cont_rep3),
                  cont_rep4 = ifelse(samples123 == 4, min(cont_rep4, na.rm = TRUE), cont_rep4)) %>%
    dplyr::mutate(treat_rep1 = ifelse(samples456 == 4, min(treat_rep1, na.rm = TRUE), treat_rep1),
                  treat_rep2 = ifelse(samples456 == 4, min(treat_rep2, na.rm = TRUE), treat_rep2),
                  treat_rep3 = ifelse(samples456 == 4, min(treat_rep3, na.rm = TRUE), treat_rep3),
                  treat_rep4 = ifelse(samples456 == 4, min(treat_rep4, na.rm = TRUE), treat_rep4))
  #
  rownames(datFiltered) = datFiltered$accession
  dat.log = datFiltered %>%
    dplyr::select(-accession, -gene, -samples123, -samples456)
  #dat.log = na.omit(datFiltered)
  #boxplot(dat.log, las = 2, main = 'total protein abundance')
  
  ##make a design table
  cond = as.factor(c(rep('cont',4),
                    rep('treat',4)))
  design = model.matrix(~0+cond)
  colnames(design) = gsub('cond', '', colnames(design))
  
  ##make the contrasts
  x = c('treat-cont')
  contrast = makeContrasts(contrasts = x, levels = design)
  fit1 = lmFit(dat.log, design)
  fit2 = contrasts.fit(fit1, contrasts = contrast)
  fit3 = eBayes(fit2)

  ##add information about PSM counts
  forPsm = processedData
  psm.count.table = data.frame(count = forPsm$n_peptides,
                               row.names = forPsm$accession)
  fit3$count = psm.count.table[rownames(fit3$coefficients),'count']
  fit4 = spectraCounteBayes(fit3)
  #VarianceBoxplot(fit4, n = 30, main = 'total proteome dataset', xlab = 'psm count')
  #VarianceScatterplot(fit4, main = 'total proteome dataset')
  
  ##extract the data
  #head(fit4$coefficients)
  psmAnnotate = dplyr::select(forPsm, accession, gene, n_fragments, n_peptides)
  valuesAnnotate = dplyr::select(datFiltered, accession, gene, samples123, samples456) %>%
    dplyr::rename(controlValues = 'samples123',
                  conditionValues = 'samples456')
  deqmsResults = tibble(outputResult(fit4, coef_col = 1)) %>%
    dplyr::rename(accession = 'gene') %>%
    dplyr::left_join(psmAnnotate) %>%
    dplyr::left_join(annotationDb) %>%
    dplyr::left_join(valuesAnnotate)
  #
  saveRDS(deqmsResults, paste(baseRepository, '/kidney/dataProcessing/dataset_deqms_', conditionsToCompare[i], '-', controlCondition, '.rds', sep = ''))
  write.table(deqmsResults, paste(baseRepository, '/kidney/dataProcessing/dataset_deqms_', conditionsToCompare[i], '-', controlCondition, '.tsv', sep = ''),
              col.names = TRUE, row.names = FALSE, sep = '\t', quote = FALSE)
  
  ##create a plot of the data
  #create the plot and save it
  goiUp = dplyr::filter(deqmsResults, logFC > 1)$gene[1:20]
  goiDn = dplyr::filter(deqmsResults, logFC < -1)$gene[1:20]
  
  plotData = deqmsResults %>%
    dplyr::mutate(logFC = ifelse(logFC > 6, 6,
                                 ifelse(logFC < -6, -6, logFC))) %>%
    dplyr::mutate(pColors = ifelse(logFC > 0.5, brewer.pal(4,'OrRd')[4],
                                   ifelse(logFC < -0.5, brewer.pal(4,'GnBu')[4], brewer.pal(3,'Greys')[2]))) %>%
    dplyr::mutate(pTextUp = ifelse(gene %in% goiUp, gene, '')) %>%
    dplyr::mutate(pTextDn = ifelse(gene %in% goiDn, gene, ''))
  #
  ggplot(plotData, aes(logFC, -log10(sca.P.Value))) +
    geom_point(size = 0.5, color = plotData$pColors) +
    labs(x = paste('log2(',conditionsToCompare[i],'-',controlCondition,')',sep=''), y = '-log10(p-value)', title = 'Differential abundance') +
    geom_text_repel(label = plotData$pTextUp, min.segment.length = unit(0, 'lines'), nudge_x = -1, nudge_y = 1, max.overlaps = 55000, size = 1, segment.size = 0.15) +
    geom_text_repel(label = plotData$pTextDn, min.segment.length = unit(0, 'lines'), nudge_x = 1, nudge_y = 1, max.overlaps = 55000, size = 1, segment.size = 0.15) +
    scale_x_continuous(limits = c(-6,6), breaks = seq(-20,20,2)) +
    scale_y_continuous(limits = c(0,20), breaks = seq(0,60,5)) +
    geom_vline(xintercept = c(-0.5,0.5), linetype = 'dashed') +
    geom_hline(yintercept = -log10(0.01), linetype = 'dashed') +
    theme_classic()
  ggsave(paste(baseRepository, '/kidney/dataProcessing/scatter_deqms_', conditionsToCompare[i], '-', controlCondition, '.pdf', sep = ''),
         height = 2, width = 2, useDingbats = FALSE)
}
```
